admin/services/data-access/pupil-restart.data.service.js

```diff
@@ -46,10 +54,100 @@ pupilRestartDataService.findLatest = async function (options) {
 
 /**
  * Get all the restart codes documents
+ * @deprecated Please use sqlFindRestartCodes instead
  * @return {Promise.<{Object}>}
  */
 pupilRestartDataService.getRestartCodes = async () => {
   return RestartCode.find().lean().exec()
 }
 
+/** SQL METHODS **/
+
+/**
+ * Create a new pupil restart.
+ * @param {object} data
+ * @return  { insertId: <number>, rowsModified: <number> }
+ */
+pupilRestartDataService.sqlCreate = async (data) => {
+  return sqlService.create(table, data)
+}
+
+/**
+ * Returns number of restarts specified by pupil id
+ * @param pupilId
+ * @return {Promise.<*>}
+ */
+pupilRestartDataService.sqlGetNumberOfRestartsByPupil = async function (pupilId) {
+  const sql = `SELECT COUNT(*) FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId`
+  const params = [
+    {
+      name: 'pupilId',
+      value: pupilId,
+      type: TYPES.Int
+    }
+  ]
+  return sqlService.query(sql, params)
+}
+
+/**
+ * Find latest check for pupil
+ * @param pupilId - the pupil taking the check
+ * @return {Promise.<void>} - lean Check objects
+ */
+pupilRestartDataService.sqlFindLatestRestart = async function (pupilId) {
+  const sql = `SELECT TOP 1 * FROM ${sqlService.adminSchema}.[pupilRestart] WHERE pupil_id=@pupilId AND isDeleted=0 ORDER BY createdAt DESC`
```

Comment:
Please could you format this sql so it is readable?

```
SELECT TOP 1 *
FROM ${sqlService.adminSchema}.[pupilRestart]
WHERE pupil_id = @pupilId AND isDeleted = 0
ORDER BY createdAt DESC
```

Reply:
@jon-shipley formatted